workflow/controller/controller_test.go

```diff
@@ -510,6 +510,13 @@ func makePodsPhase(ctx context.Context, woc *wfOperationCtx, phase apiv1.PodPhas
 			if err != nil {
 				panic(err)
 			}
+			if phase == apiv1.PodSucceeded {
+				if woc.wf.Status.TaskResultsCompletionStatus == nil {
```

Comment:
`workflow_types.go` has `MarkTaskResultIncomplete` and `markTaskResultComplete` convenience methods that handle initialization of `TaskResultsCompletionStatus`.

Reply:
Yes，I can change it.